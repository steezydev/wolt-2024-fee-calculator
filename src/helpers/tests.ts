const localStorageMock = (function () {
    let store: Record<string, string> = {};
    return {
        getItem(key: string) {
            return store[key] || null;
        },
        setItem(key: string, value: string) {
            store[key] = value.toString();
        },
        clear() {
            store = {};
        },
    };
})();

const matchMediaMock = jest.fn().mockImplementation((query) => ({
    matches: query === '(prefers-color-scheme: dark)',
    media: query,
    onchange: null,
    addListener: jest.fn(),
    removeListener: jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
}));

export const mockContext = () => {
    beforeAll(() => {
        Object.defineProperty(window, 'localStorage', {
            value: {
                getItem: jest.fn().mockImplementation((key: string) => {
                    return localStorageMock.getItem(key);
                }),
                setItem: jest
                    .fn()
                    .mockImplementation((key: string, value: string) => {
                        localStorageMock.setItem(key, value);
                    }),
                clear: localStorageMock.clear,
            },
            writable: true,
        });

        window.matchMedia = matchMediaMock;
    });

    beforeEach(() => {
        (window.localStorage.getItem as jest.Mock).mockClear();
        (window.localStorage.setItem as jest.Mock).mockClear();
        matchMediaMock.mockClear();
    });
};

export const setupFakeDate = () => {
    beforeEach(() => {
        jest.useFakeTimers();
        jest.setSystemTime(new Date('2024-01-01'));
    });

    afterEach(() => {
        jest.useRealTimers();
    });
};

export const mockScrollTo = () => {
    beforeEach(() => {
        Element.prototype.scrollTo = jest.fn();
    });

    afterEach(() => {
        (Element.prototype.scrollTo as jest.Mock).mockRestore();
    });
};
