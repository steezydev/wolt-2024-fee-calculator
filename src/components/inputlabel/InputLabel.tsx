import AlertOctagonIcon from '@/components/icons/AlertOctagonIcon';
import B3 from '@/components/typography/B3';
import { classNames } from '@/helpers/classNames';
import { withClassNames } from '@/hoc/StyleHoc';
import { InputLabelProps } from '@/types/props/InputLabelProps';

const InputLabel = ({
  label,
  id,
  children,
  errorMessage,
  className,
}: InputLabelProps) => {
  return (
    <div className={classNames('group flex flex-col gap-1', className)}>
      {withClassNames(children, 'peer order-2')}
      <label
        className={classNames(
          'text-neutral-light-300 dark:text-neutral-dark-300 text-sm font-semibold order-1',
          'peer-focus:text-primary-300 peer-data-[invalid=true]:text-error peer-data-[focused=true]:peer-data-[invalid=false]:text-primary-300'
        )}
        htmlFor={id}
      >
        {label}
      </label>
      <div className='hidden order-3 peer-data-[invalid=true]:flex gap-1.5 items-center'>
        <AlertOctagonIcon className='w-5 h-5 text-error' />
        <B3 id={id + 'ErrorMessage'} className='text-error'>
          {errorMessage}
        </B3>
      </div>
    </div>
  );
};

export default InputLabel;
