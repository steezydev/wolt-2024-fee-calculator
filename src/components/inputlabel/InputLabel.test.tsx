import InputLabel from './InputLabel';
import { inputLabelRenderTests } from './InputLabel.tests';

inputLabelRenderTests('InputLabel', InputLabel);
