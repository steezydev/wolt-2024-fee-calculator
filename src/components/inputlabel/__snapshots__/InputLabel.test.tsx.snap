// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputLabel when asked to render with additional class  renders with additional class 1`] = `
<div
  className="group flex flex-col gap-1 some-class"
>
  <input
    aria-invalid={false}
    aria-label="Test label"
    autoFocus={false}
    className="[appearance:textfield] [&::-webkit-outer-spin-button]:appearance-none [&::-webkit-inner-spin-button]:appearance-none rounded-xl bg-white border-2 border-neutral-light-100 placeholder:text-neutral-light-300 px-3 py-3 text-xl text-black dark:text-white leading-none group-hover:valid:border-primary-300 invalid:border-error  focus:outline-none focus:ring-1 focus:ring-primary-300 focus:border-primary-300 peer order-2"
    data-test-id="test-input"
    disabled={false}
    id="test-input"
    name="test-input"
    onChange={[Function]}
    placeholder="20"
    required={true}
    type="number"
    value={20}
  />
  <label
    className="text-neutral-light-300 dark:text-neutral-dark-300 text-sm font-semibold order-1 peer-focus:text-primary-300 peer-invalid:text-error"
    htmlFor="test-input"
  >
    Test label
  </label>
</div>
`;

exports[`InputLabel when asked to render with label  renders with label 1`] = `
<div
  className="group flex flex-col gap-1"
>
  <input
    aria-invalid={false}
    aria-label="Test label"
    autoFocus={false}
    className="[appearance:textfield] [&::-webkit-outer-spin-button]:appearance-none [&::-webkit-inner-spin-button]:appearance-none rounded-xl bg-white border-2 border-neutral-light-100 placeholder:text-neutral-light-300 px-3 py-3 text-xl text-black dark:text-white leading-none group-hover:valid:border-primary-300 invalid:border-error  focus:outline-none focus:ring-1 focus:ring-primary-300 focus:border-primary-300 peer order-2"
    data-test-id="test-input"
    disabled={false}
    id="test-input"
    name="test-input"
    onChange={[Function]}
    placeholder="20"
    required={true}
    type="number"
    value={20}
  />
  <label
    className="text-neutral-light-300 dark:text-neutral-dark-300 text-sm font-semibold order-1 peer-focus:text-primary-300 peer-invalid:text-error"
    htmlFor="test-input"
  >
    Test label
  </label>
</div>
`;
